crates/nu-command/src/math/utils.rs

```diff
@@ -94,3 +94,86 @@ pub fn calculate(
         )),
     }
 }
+
+// Utility functions to reduce boiler plate in the the
+// trigonometric math commands
+pub fn trig_value<F>(value: Value, use_degrees: Option<bool>, f: F) -> Value
+where
+    F: Fn(f64) -> f64,
+{
+    let (val, span) = match value {
+        Value::Int { val, span } => (val as f64, span),
+        Value::Float { val, span } => (val, span),
+        _ => unreachable!(),
+    };
+
+    let use_degrees = use_degrees.unwrap_or(false);
+
+    let val = if use_degrees { val.to_radians() } else { val };
+
+    Value::Float { val: f(val), span }
+}
+
+pub fn trig_error(other: Value, head: Span) -> Value {
```

Comment:
The naming of this helper is misleading as it is neither specific to trigonometry nor describes the particular reason for using it.
```suggestion
pub fn math_type_error(other: Value, head: Span) -> Value {
```